backend/ethereum/channel/errors.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package channel
+
+import "strings"
+
+type (
+	// OnChainTxType defines the type of on-chain transaction function names that
+	// can be returned in TxTimeoutError.
+	OnChainTxType int
+)
+
+// Enumeration of valid transaction types.
+//
+// Fund funds the channel for a given user.
+// Register registers a state of the channel. The state be concluded after the challenge duration has passed.
+// Progress progresses the state of the channel directly on the blockchain.
+// Conclude concludes the state of a channel after it had been registered and the challenge duration has passed.
+// ConcludeFinal directly concludes the finalized state of the channel without registering it.
+// Withdraw withdraws the funds for a given user after the channel was concluded.
+const (
+	Fund OnChainTxType = iota
+	Register
+	Progress
+	Conclude
+	ConcludeFinal
+	Withdraw
+)
+
+var onChainTxTypeNames = [...]string{
+	"fund",
+	"register",
+	"progress",
+	"conclude",
+	"concludeFinal",
+	"withdraw",
+}
+
+func (t OnChainTxType) String() string {
+	return onChainTxTypeNames[t]
+}
+
```

Comment:
@manoranjith you could add the `checkIsChainNotReachableError` function here, since it reads a bit weird when you define it locally so often.

Reply:
To me, moving the error checking logic to a separate function seemed to interrupt the readability. That is why I defined as local function in multiple places, though it was duplication.

If we can add the `checkIsChainNotReachableError` as a function, then can we not simply use this function in all places where we make this check ? @ggwpez

Reply:
Fine either way for me, maybe @matthiasgeihs wants to take another look.

Reply:
I think we can move it in one place and avoid duplication, as oli suggested.

Reply:
Okay, I will update the PR by defining `checkIsChainNotReachableError` and using it in all places.

Reply:
@matthiasgeihs @ggwpez I have updated made the changes in these two commits (7f0d8ee and  9a325ce).

Can you have a look and see if  this is fine. If is it, then I will squash them into the previous commits.

Reply:
Thanks looks good, then please squash.

Reply:
Done.